### PR TITLE
adds public api bulk upload support

### PR DESCRIPTION
**Pull request by darwayne**
### Background

Adds bulk upload support via the api token

### Changes

* Adds real implementation for the public api client around bulk uploads

### CODE REVIEW

File: panther_analysis_tool/backend/public_api_client.py [hunk 1]
```diff
@@ -120,8 +150,8 @@ def list_managed_schema_updates(self) -> BackendResponse:
     def update_managed_schemas(self, params: UpdateManagedSchemasParams) -> BackendResponse:
         pass
 
-    def _execute(self, request: DocumentNode) -> ExecutionResult:
-        return self._gql_client.execute(request, get_execution_result=True)
+    def _execute(self, request: DocumentNode, variable_values: Optional[Dict[str, Any]] = None, ) -> ExecutionResult:
```
Comment (danbiwer) [inline panther_analysis_tool/backend/public_api_client.py:153]: cool, this is exactly what I did in my branch

File: panther_analysis_tool/backend/public_api_client.py [hunk 2]
```diff
@@ -106,7 +110,33 @@ def check(self) -> BackendCheckResponse:
...
+        query = self._requests.bulk_upload_mutation()
+        upload_params = {"input": {"data": params.encoded_bytes()}}
+        default_stats = dict(total=0, new=0, modified=0)
+        res = res = self._execute(query, variable_values=upload_params)
+
+        if res.errors:
+            for err in res.errors:
+                logging.error(err.message)
+
+            return BackendCheckResponse(
```
Comment (nmartin413) [inline panther_analysis_tool/backend/public_api_client.py:122]: `BackendCheckResponse` was meant to be a special value that only came from `def check()`. We could probably eliminate it and use the generic `BackendResponse[CheckResponse]` format instead 

Might want to extend `BackendResponse` to have an error field?

Comment (darwayne) [reply panther_analysis_tool/backend/public_api_client.py:122]: woah good catch

**Review by danbiwer — APPROVED** [review]
awesome 💯

**Pull request merged**